Tasks/AzureCLIV1/azureclitask.ts

```diff
@@ -135,10 +135,27 @@ export class azureclitask {
         //login using svn
         this.throwIfError(tl.execSync("az", "login --service-principal -u \"" + servicePrincipalId + "\" -p \"" + cliPassword + "\" --tenant \"" + tenantId + "\""), tl.loc("LoginFailed"));
         this.isLoggedIn = true;
+        
+        // create config.json 
+        this.createServicePrincipalsIfRequired(servicePrincipalId, cliPassword, subscriptionID);
+
         //set the subscription imported to the current subscription
         this.throwIfError(tl.execSync("az", "account set --subscription \"" + subscriptionID + "\""), tl.loc("ErrorInSettingUpSubscription"));
     }
 
+    private static createServicePrincipalsIfRequired(servicePrincipalId, client_key, subscriptionID): void {
+        var shouldCreateServicePrincipalJSONfile = tl.getVariable("CREATE_SERVICE_PRINCIPAL_FILE");
+        if (!!shouldCreateServicePrincipalJSONfile) {
+            var fileName = "aksServicePrincipal.json";
+            fs.writeFileSync(
+                path.join(fileName),
+                JSON.stringify({ [subscriptionID]: { client_secret: client_key, service_principal: servicePrincipalId } })
+            );
+
+            fs.link(fileName, path.join(this.azCliConfigPath, fileName));
```

Comment:
Why are you not creating the file in the eventual location path.join(this.azCliConfigPath, fileName)  itself?
```suggestion
            fs.writeFileSync(path.join(this.azCliConfigPath, fileName), JSON.stringify..);
```

Reply:
`writeFileSync` fails to create a new file if it is outside the application directory. And azCliConfigPath is outside it, so I had to create it here, and move it there.

Reply:
Just for my knowledge, does link method move the file, or creates a link? if it just links, we would need to delete the original file as well.